voluptuous/humanize.py

```diff
@@ -12,6 +12,8 @@ def _nested_getitem(data, path):
         except (KeyError, IndexError):
             # The index is not present in the dictionary, list or other indexable
             return None
+        except TypeError: # data is not subscriptable
+            break
```

Comment:
The advantage of resolving until you hit a failure (by breaking out) is that it helps you in the correct direction.

Happy to comply either way, your call